metricq/types.py

```diff
@@ -264,10 +288,10 @@ def __sub__(self, other: Union["Timedelta", "Timestamp", datetime.timedelta]):
             "invalid type to subtract from Timedelta: {}".format(type(other))
         )
 
-    def __truediv__(self, factor):
+    def __truediv__(self, factor) -> "Timedelta":
```

Comment:
No typing for factor?

Reply:
Well... this should be `float`, right?

Reply:
- The internal type should always be an `int`
- We should allow `float` and `int` as parameter (type `float`)
- The previous point should be tested.

Reply:
I guess this will be its own PR. I found a few more things that might need refactoring and are related to `Timedelta`s.